pkg/network/passt/repair_test.go

```diff
@@ -0,0 +1,211 @@
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			),
+		),
+		Entry("When an iface connected to pod network uses tap attachment",
+			libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceWithBindingPlugin("default", v1.PluginBinding{Name: "tap"})),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			),
+		),
+		Entry("When an iface connected to pod network uses managedTap attachment",
+			libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceWithBindingPlugin("default", v1.PluginBinding{Name: "managedTap"})),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			),
+		),
+		Entry("When there is no iface connected to pod network",
+			libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding("secondary")),
+				libvmi.WithNetwork(libvmi.MultusNetwork("secondary", "secondary-nad")),
+			),
+		),
+	)
+
+	DescribeTable("Should run passt repair", func(vmi *v1.VirtualMachineInstance) {
+		clusterConfig := stubClusterConfig{
+			registeredPlugins: map[string]v1.InterfaceBindingPlugin{
+				"passt": {
+					SidecarImage:                "passt:latest",
+					NetworkAttachmentDefinition: "default/passt-network",
+				},
+			},
+		}
+
+		handler := passt.NewRepairManagerWithOptions(
+			clusterConfig,
+			fakeCreateShortenedSymlink,
+			fakeFindRepairSocketInDir,
+			fakeCommand,
+			newStubStore(),
+		)
+		Expect(handler.HandleMigrationSource(vmi, stubSocketDirFunc)).To(Succeed())
+	},
+		Entry("When an iface is connected to pod network using passt binding plugin",
+			libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceWithPasstBindingPlugin()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			),
+		),
+		Entry("When an iface is connected to Multus default network using passt binding plugin",
+			libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceWithPasstBindingPlugin()),
+				libvmi.WithNetwork(&v1.Network{
+					Name: "default",
+					NetworkSource: v1.NetworkSource{
+						Multus: &v1.MultusNetwork{
+							NetworkName: "alternative",
+							Default:     true,
+						},
+					},
+				}),
+			),
+		),
+	)
+
+	It("Should return error when socketDirFunc returns an error", func() {
+		clusterConfig := stubClusterConfig{
+			registeredPlugins: map[string]v1.InterfaceBindingPlugin{
+				"passt": {
+					SidecarImage:                "passt:latest",
+					NetworkAttachmentDefinition: "default/passt-network",
+				},
+			},
+		}
+
+		handler := passt.NewRepairManagerWithOptions(
+			clusterConfig,
+			fakeCreateShortenedSymlink,
+			fakeFindRepairSocketInDir,
+			fakeCommand,
+			newStubStore(),
+		)
+
+		vmi := libvmi.New(
+			libvmi.WithInterface(libvmi.InterfaceWithPasstBindingPlugin()),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+		)
+
+		expectedError := errors.New("socketDirFunc returns an error")
+
+		failingSocketDirFunc := func(_ *v1.VirtualMachineInstance) (string, error) {
+			return "", expectedError
+		}
+
+		Expect(handler.HandleMigrationSource(vmi, failingSocketDirFunc)).To(MatchError(expectedError))
+	})
```

Comment:
Please add a test case asserting passt-repair should not run, because it is already running.